qucs/settings.cpp

```diff
@@ -0,0 +1,75 @@
+#include <QApplication>
+
+#include "main.h"
+#include "settings.h"
+#include "extsimkernels/spicecompat.h"
+
+settingsManager::settingsManager()
+    :QSettings("qucs", "qucs_s")
+{
+    // qDebug() << this << " created " << organizationName() << " " << applicationName();
+
+    initAliases();
+    initDefaults();
+}
+
+settingsManager::~settingsManager()
+{
+    // qDebug() << this << " destroyed";
+}
+
+void settingsManager::resetDefaults(const QString &group)
+{
+    qDebug() << "Reset settings group " << group;
+
+    if (group == "All") {
+        // Remove all settings (including those for which there is no default).
+        // clear();
+
+        // Repopulate with known defaults.
+        for (auto const& item : m_Defaults) {
+            qDebug() << "Resetting item " << item.first << " " << item.second;
+            setValue(item.first, item.second);
+        }
+    }
+
+    else {
+        beginGroup(group);
+        for (const QString& key : allKeys()) {
+            setValue(key, m_Defaults[key]);
+        }
+    }
+}
+
+void settingsManager::initDefaults()
+{
+    m_Defaults["DefaultSimulator"] = spicecompat::simNotSpecified;
+    m_Defaults["FirstRun"] = true;
+    m_Defaults["font"] = QApplication::font();
+    m_Defaults["appFont"] = QApplication::font();
+    m_Defaults["LargeFontSize"] = static_cast<double>(16.0);
+    m_Defaults["maxUndo"] = 20;
+    m_Defaults["panelIconsTheme"] = qucs::autoIcons;
+    m_Defaults["compIconsTheme"] = qucs::autoIcons;
+
+#ifdef Q_OS_WIN
+    m_Defaults["NgspiceExecutable"] = "ngspice_con.exe";
+    m_Defaults["XyceExecutable"] = "Xyce.exe";
+#else
+    m_Defaults["NgspiceExecutable"] = "ngspice.exe";
```

Comment:
`ngspice` shoud be here without `exe` suffix.

Reply:
Oops!